modules/heat_conduction/src/fvbcs/FunctorThermalResistanceBC.C

```diff
@@ -0,0 +1,183 @@
+  return params;
+}
+
+FunctorThermalResistanceBC::FunctorThermalResistanceBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _geometry(getParam<MooseEnum>("geometry").getEnum<Moose::CoordinateSystemType>()),
+    _inner_radius(
+        _geometry == Moose::CoordinateSystemType::COORD_RZ ? getParam<Real>("inner_radius") : 1.0),
+    _T(isParamValid("temperature") ? getFunctor<ADReal>("temperature")
+                                   : getFunctor<ADReal>("variable")),
+    _T_ambient(getParam<Real>(HeatConduction::T_ambient)),
+    _k(getParam<std::vector<Real>>("thermal_conductivities")),
+    _dx(getParam<std::vector<Real>>("conduction_thicknesses")),
+    _h(getFunctor<ADReal>(getParam<MooseFunctorName>("htc"))),
+    _emissivity(getParam<Real>(HeatConduction::emissivity)),
+    _max_iterations(getParam<unsigned int>("max_iterations")),
+    _tolerance(getParam<Real>("tolerance")),
+    _alpha(getParam<Real>("step_size")),
+    _T_surface(0.0),
+    _outer_radius(_inner_radius),
+    _conduction_resistance(0.0),
+    _parallel_resistance(0.0)
+{
+  if (_k.size() != _dx.size())
+    paramError("conduction_thicknesses",
+               "Number of specified thermal conductivities must match "
+               "the number of conduction layers!");
+
+  if (_geometry == Moose::CoordinateSystemType::COORD_RZ)
+    for (const auto & d : _dx)
+      _outer_radius += d;
+
+  // because the thermal conductivities are constant, we only need to compute
+  // the conduction resistance one time
+  computeConductionResistance();
+}
+
+void
+FunctorThermalResistanceBC::computeConductionResistance()
+{
+  Real r = _inner_radius;
+
+  for (const auto i : index_range(_k))
+  {
+    switch (_geometry)
+    {
+      case Moose::CoordinateSystemType::COORD_XYZ:
+        _conduction_resistance += _dx[i] / _k[i];
+        break;
+      case Moose::CoordinateSystemType::COORD_RZ:
+      {
+        _conduction_resistance += std::log((_dx[i] + r) / r) / _k[i];
+        r += _dx[i];
+        break;
+      }
+      default:
+        mooseError("Unhandled 'GeometryEnum' in 'FunctorThermalResistanceBC'!");
+    }
+  }
+}
+
+ADReal
+FunctorThermalResistanceBC::computeQpResidual()
+{
+  // Evaluate material properties on the face
+  const auto face_arg = singleSidedFaceArg();
+
+  // radiation resistance has to be solved iteratively, since we don't know the
+  // surface temperature. We do know that the heat flux in the conduction layers
+  // must match the heat flux in the parallel convection-radiation segment. For a
+  // first guess, take the surface temperature as the average of _T and T_ambient.
+  _T_surface = 0.5 * (_T(face_arg) + _T_ambient);
+
+  // total flux perpendicular to boundary
+  ADReal flux;
+
+  // resistance component representing the sum of the convection and radiation
+  // resistances, in parallel
+  computeParallelResistance();
+
+  // other iteration requirements
+  Real iteration = 0;
```

Comment:
```suggestion
  unsigned int iteration = 0;
```

Reply:
how was I not getting a warning for the comparison to _max_iterations (an unsigned int) for this